src/core/Aggregators.java

```diff
@@ -332,4 +406,90 @@ public Interpolation interpolationMethod() {
     
   }
 
+  /**
+   * Percentile aggregator based on apache commons math implementation
+   */
+  private static final class Percentile implements Aggregator {
+    private final Float percentile;
+    private final String name;
+
+    public Percentile(final Float percentile, final String name) {
+      Preconditions.checkArgument(percentile > 0 && percentile <= 100.d, "Invalid percentile value");
+      this.percentile = percentile;
+      this.name = name;
+    }
+
+    public long runLong(final Longs values) {
+      DescriptiveStatistics stats = new DescriptiveStatistics();
```

Comment:
Instead of ``DescriptiveStats``, use the ``Percentile`` class since it has less overhead. Try something like:
```
    final Percentile evalulator = new Percentile(percentile)
      .withEstimationType(estimation_type);
    final ResizableDoubleArray local_values = new ResizableDoubleArray();
    while(vs.hasNextValue()) {
      local_values.addElement(vs.nextDoubleValue());
    }
    evalulator.setData(local_values.getElements());
    return evalulator.evaluate();
```

The default for the ``P99``, ``P95`` etc. aggregators should be ``EstimationType.R_3`` since that will look at the entire set and pick out the closest value. For the ``EP99``, etc, maybe pick one of the other types from http://en.wikipedia.org/wiki/Quantile. Then we don't need to import Matt Proud's library until it makes it into Apache Commons or another well used repo.